source/loader/layers/sanitizer/asan_interceptor.cpp

```diff
@@ -55,80 +83,34 @@ uptr MemToShadow_PVC(uptr USM_SHADOW_BASE, uptr UPtr) {
-        nullptr);
-    assert(Result == UR_RESULT_SUCCESS);
-    return Context;
-}
-
-ur_device_handle_t getDevice(ur_queue_handle_t Queue) {
-    ur_device_handle_t Device;
-    [[maybe_unused]] auto Result = context.urDdiTable.Queue.pfnGetInfo(
-        Queue, UR_QUEUE_INFO_DEVICE, sizeof(ur_device_handle_t), &Device,
-        nullptr);
-    assert(Result == UR_RESULT_SUCCESS);
-    return Device;
-}
-
-ur_program_handle_t getProgram(ur_kernel_handle_t Kernel) {
-    ur_program_handle_t Program;
-    [[maybe_unused]] auto Result = context.urDdiTable.Kernel.pfnGetInfo(
-        Kernel, UR_KERNEL_INFO_PROGRAM, sizeof(ur_program_handle_t), &Program,
-        nullptr);
-    assert(Result == UR_RESULT_SUCCESS);
-    return Program;
-}
-
-void getProgramDevices(ur_program_handle_t Program,
-                       std::vector<ur_device_handle_t> &Devices) {
-    size_t PropSize;
-    [[maybe_unused]] ur_result_t Result = context.urDdiTable.Program.pfnGetInfo(
-        Program, UR_PROGRAM_INFO_DEVICES, 0, nullptr, &PropSize);
-    assert(Result == UR_RESULT_SUCCESS);
-
-    Devices.resize(PropSize / sizeof(ur_device_handle_t));
-    Result = context.urDdiTable.Program.pfnGetInfo(
-        Program, UR_PROGRAM_INFO_DEVICES, PropSize, Devices.data(), nullptr);
-    assert(Result == UR_RESULT_SUCCESS);
-}
-
-size_t getLocalMemorySize(ur_device_handle_t Device) {
-    size_t LocalMemorySize;
-    [[maybe_unused]] auto Result = context.urDdiTable.Device.pfnGetInfo(
-        Device, UR_DEVICE_INFO_LOCAL_MEM_SIZE, sizeof(LocalMemorySize),
-        &LocalMemorySize, nullptr);
-    assert(Result == UR_RESULT_SUCCESS);
-    return LocalMemorySize;
-}
-
-std::string getKernelName(ur_kernel_handle_t Kernel) {
-    size_t KernelNameSize = 0;
-    [[maybe_unused]] auto Res = context.urDdiTable.Kernel.pfnGetInfo(
-        Kernel, UR_KERNEL_INFO_FUNCTION_NAME, 0, nullptr, &KernelNameSize);
-    assert(Res == UR_RESULT_SUCCESS);
-
-    std::vector<char> KernelNameBuf(KernelNameSize);
-    Res = context.urDdiTable.Kernel.pfnGetInfo(
-        Kernel, UR_KERNEL_INFO_FUNCTION_NAME, KernelNameSize,
-        KernelNameBuf.data(), nullptr);
-    assert(Res == UR_RESULT_SUCCESS);
-
-    return std::string(KernelNameBuf.data(), KernelNameSize - 1);
-}
-
 } // namespace
 
-SanitizerInterceptor::SanitizerInterceptor()
-    : m_IsInASanContext(IsInASanContext()),
-      m_ShadowMemInited(m_IsInASanContext) {}
+SanitizerInterceptor::SanitizerInterceptor() {
+    auto Options = getenv_to_map("UR_LAYER_ASAN_OPTIONS");
+    if (!Options.has_value()) {
+        return;
+    }
+    auto KV = Options->find("debug");
+    if (KV != Options->end()) {
+        auto Value = KV->second.front();
+        if (Value == "1" || Value == "true") {
+            cl_Debug = 1;
+        }
+    }
+    KV = Options->find("quarantine_size_mb");
+    if (KV != Options->end()) {
+        auto Value = KV->second.front();
+        cl_MaxQuarantineSizeMB = std::stoul(Value);
```

Comment:
stoul may throw exception, is it catched? please also see std::make_unique below.

Reply:
done.
it seems like "std::make_unique" is ignored by other code as well.